design/src/main/java/com/urlaunched/android/design/ui/progressbarcontainer/ProgressBarContainer.kt

```diff
@@ -45,6 +48,59 @@ fun ProgressBarContainer(
     }
 }
 
+@Composable
+fun ProgressBarContainer(
+    modifier: Modifier = Modifier,
+    isLoading: Boolean,
+    indicatorColor: Color = ProgressIndicatorDefaults.circularColor,
+    indicatorTrackColor: Color = ProgressIndicatorDefaults.circularTrackColor,
+    progress: Float = 0F,
+    progressSettings: ProgressSettings,
+    content: @Composable () -> Unit
+) {
+    Box(modifier = modifier.fillMaxSize()) {
+        content()
+
+        if (isLoading) {
+            Box(
+                modifier = Modifier
+                    .fillMaxSize()
+                    .background(Color.Black.copy(alpha = 0.75f))
+                    .clickable(enabled = false, onClick = {}),
+                contentAlignment = Alignment.Center
+            ) {
+                if (progress > 1) {
+                    CircularProgressIndicator(
+                        modifier = Modifier.size(50.dp),
+                        color = indicatorColor,
+                        trackColor = indicatorTrackColor
+                    )
+                } else {
+                    CircularProgressIndicator(
+                        modifier = Modifier.size(50.dp),
+                        progress = { progress },
+                        color = indicatorColor,
+                        trackColor = indicatorTrackColor
+                    )
+                    Text(
+                        text = "${(progress * 100).toInt()}%",
+                        style = progressSettings.mainTextStyle,
+                        color = progressSettings.mainColor
+                    )
+                    if (progressSettings.description?.isNotEmpty() == true) {
+                        Text(
+                            text = progressSettings.description,
+                            modifier = Modifier.padding(top = 90.dp),
+                            style = progressSettings.descriptionTextStyle,
+                            color = progressSettings.descriptionColor
+                        )
+                    }
```

Comment:
What do you think about moving these elements into separate composable component and adding a composable lambda argument as a way to customize the element without the need to update the library? Maybe it's even better to leave this ProgressBarContainer arguments as is and create separate overload function for this and have current ProgressBarContainer wrap the other one

Reply:
I'm afraid I'm not sure what exactly are your proposals. Can you write some pseudocode? Just to have a better idea. It doesn't have to work properly.

Reply:
Something along the lines of 
```
@Composable
fun ProgressBarContainer(
    modifier: Modifier = Modifier,
    isLoading: Boolean,
    indicatorColor: Color = ProgressIndicatorDefaults.circularColor,
    indicatorTrackColor: Color = ProgressIndicatorDefaults.circularTrackColor,
    progress: Float = 0F,
    progressSettings: ProgressSettings,
    progressContent: @Composable () -> Unit
    content: @Composable () -> Unit
) {
    Box(modifier = modifier.fillMaxSize()) {
        content()

        if (isLoading) {
            Box(
                modifier = Modifier
                    .fillMaxSize()
                    .background(Color.Black.copy(alpha = 0.75f))
                    .clickable(enabled = false, onClick = {}),
                contentAlignment = Alignment.Center
            ) {
                if (progress > 1) {
                    CircularProgressIndicator(
                        modifier = Modifier.size(50.dp),
                        color = indicatorColor,
                        trackColor = indicatorTrackColor
                    )
                } else {
                    CircularProgressIndicator(
                        modifier = Modifier.size(50.dp),
                        progress = { progress },
                        color = indicatorColor,
                        trackColor = indicatorTrackColor
                    )
                    
                    progressContent()

// Rest of the code

```